### PR TITLE
[Storage] Fix parser logic when downloading a managed disk using SAS and bearer token 

### PR DESCRIPTION
**Pull request by yifanz7**
## Description
Fix https://github.com/Azure/azure-powershell/issues/24104

## Mandatory Checklist

- Please choose the target release of Azure PowerShell
  - [x] [General release](../blob/main/CONTRIBUTING.md#target-release-types)
  - [ ] [Public preview](../blob/main/CONTRIBUTING.md#target-release-types)
  - [ ] [Private preview](../blob/main/CONTRIBUTING.md#target-release-types)
  - [ ] [Engineering build](../blob/main/CONTRIBUTING.md#target-release-types)
  - [ ] N/A

- [ ] Check this box to confirm: **I have read the [_Submitting Changes_](../blob/main/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](../blob/main/CONTRIBUTING.md) and reviewed the following information:**

* **SHOULD** update `ChangeLog.md` file(s) appropriately
    * For SDK-based development mode, update `src/{{SERVICE}}/{{SERVICE}}/ChangeLog.md`.
        * A snippet outlining the change(s) made in the PR should be written under the `## Upcoming Release` header in the past tense. 
    * For autorest-based development mode, include the changelog in the PR description.
    * Should **not** change `ChangeLog.md` if no new release is required, such as fixing test case only.
* **SHOULD** regenerate markdown help files if there is cmdlet API change. [Instruction](../blob/main/documentation/development-docs/help-generation.md#updating-all-markdown-files-in-a-module)
* **SHOULD** have proper test coverage for changes in pull request.
* **SHOULD NOT** adjust version of module manually in pull request

### CODE REVIEW

File: src/Storage/Storage/Blob/Cmdlet/GetAzureStorageBlobContent.cs [hunk 1]
```diff
@@ -433,6 +433,38 @@ internal string GetFullReceiveFilePath(string fileName, string blobName, DateTim
...
+        {
+            string authenticateHeaderName = "WWW-Authenticate";
+            string audience = null;
+            foreach (var header in exception.GetRawResponse().Headers)
+            {
+                string headerName = header.Name;
+                if (headerName.StartsWith(authenticateHeaderName))
+                {
+                    string headerValue = header.Value;
+                    WriteDebugLog(string.Format("Found header: {0}", headerValue));
```
Comment (blueww) [inline src/Storage/Storage/Blob/Cmdlet/GetAzureStorageBlobContent.cs:451]: It's better to also add the header name.

Comment (yifanz7) [reply src/Storage/Storage/Blob/Cmdlet/GetAzureStorageBlobContent.cs]: Added

File: src/Storage/Storage/Blob/Cmdlet/GetAzureStorageBlobContent.cs [hunk 2]
```diff
@@ -433,6 +433,38 @@ internal string GetFullReceiveFilePath(string fileName, string blobName, DateTim
...
+                    string audienceName = "resource_id=";
+                    try
+                    {
+                        audience = headerValue.Split(new string[] { audienceName }, StringSplitOptions.None)[1];
```
[the remark this answers is not in the document]
Comment (yifanz7) [reply src/Storage/Storage/Blob/Cmdlet/GetAzureStorageBlobContent.cs]: Done

**Review by blueww — APPROVED** [review]
As I know, Yifan has tested the fix code on Mac/Linux (PSH 7) and window (both PSH 5 and PSH 7). @yifanz7  to confirm.

**Review by msJinLei — APPROVED** [review]

**Comment by yifanz7** [comment]
> As I know, Yifan has tested the fix code on Mac/Linux (PSH 7) and window (both PSH 5 and PSH 7). @yifanz7 to confirm.

Yes correct

**Pull request merged**